install.xml

```diff
@@ -258,13 +258,19 @@
             if($validated_collivery_data){
                 $data['quote_data'] = $validated_collivery_data;
                 $check = $this->db->query('select waybill_id from ' .DB_PREFIX . 'order where order_id="' . $data_object->order_id . '"')->row['waybill_id'];
-                if(!$check){
-                    $waybill_id = $this->collivery->addCollivery($validated_collivery_data);
-                    if($waybill_id){
+                if(!$check) {
+                    try {
+                        $waybill_id = $this->collivery->addCollivery($validated_collivery_data);
                         $this->db->query('update ' .DB_PREFIX . "order set `collivery_price_data`='". json_encode($validated_collivery_data)."', waybill_id='" . $waybill_id . "' where order_id='" . $data_object->order_id . "'");
                         if($this->config->get('shipping_mds_auto_accept_waybill')){
                             $this->collivery->acceptCollivery($waybill_id);
                         }
+                    } catch (\Mds\ColliveryException $e) {
+                        $this->response->addHeader($this->request->server['SERVER_PROTOCOL'] . '/1.1 400 Bad Request');
```

Comment:
can you assume all are 400 errors?

Reply:
That's very good point, no we can't it could very well be an error in the Collivery API.
*however*, given the crude frontend implementation that is consuming this (`$.ajax()`) -
It doesn't make a difference what the error code is - as long as the status is an error  and `jQuery` enters the `fail()` method of the ajax call.